File: current_session/169.java
class Solution {
    public int majorityElement(int[] nums) {
        int pick = nums[0];
        int cnt = 1;
        for (int i=1; i<nums.length; i++) {
            if (cnt == 0) {
                pick = nums[i];
                cnt = 1;
            } else if (nums[i] == pick) {
                cnt++;
            } else {
                cnt--;
            }
        }
        return pick;
    }
}

File: current_session/python/169.py
import collections
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        # return collections.Counter(nums).most_common(1)[0][0]
        pick, cnt = nums[0], 1
        for i in range(1, len(nums)):
            if cnt == 0:
                cnt, pick = cnt+1, nums[i]
            elif nums[i] == pick:
                cnt += 1
            else:
                cnt -= 1
        return pick